Develop

firuzaminov opened the pull request:
Аминов Фируз, финальный проект 4-го спринта

firuzaminov committed:
Second commit

firuzaminov committed:
Third commit

firuzaminov committed:
Delete target directory

firuzaminov committed:
Delete .idea directory

nikolay-977 commented on src/main/java/ru/yandex/prakrikum/MainPageScooter.java:
```diff
@@ -0,0 +1,121 @@
+    // Локатор для восьмого вопроса "Я жизу за МКАДом, привезёте?"
+    private static final By I_LIVE_OUTSIDE_THE_MOSCOW = By.className("accordion__item");
+    private static final By DOWN_ORDER_BUTTON = By.className("Home_FinishButton__1_cWm");
+
```
Необходимо исправить: убери лишние пропуски строк. Для зрительного разделения кода достаточно использовать один пропуск строки. Пропуски строк должны быть однообразными.

nikolay-977 commented on src/main/java/ru/yandex/prakrikum/MainPageScooter.java:
```diff
@@ -0,0 +1,121 @@
+        _header = new HeaderWithTwoButtons(driver);
+    }
+    // Метод для нажатия на галочку и проверки выпадающего текста
+    public void scrollInformationAndShowText(){
```
Необходимо исправить: для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea используй сочетание клавиш Ctrl+Alt+L. Также можно установить курсор на корневую директорию и нажать клавиш Ctrl+Alt+L, отметить чек-боксы: cleanup code, а также optimize imports(для оптимизации импортов) и rearrange entries(для изменения порядка переменных и методов в соответствии с best practics)

nikolay-977 commented on src/main/java/ru/yandex/prakrikum/MainPageScooter.java:
```diff
@@ -0,0 +1,121 @@
+        driver.findElements(HOW_MUCH_AND_PAY_LIST).get(0).click();
+        new WebDriverWait(driver, Duration.ofSeconds(3)).until(ExpectedConditions.visibilityOfElementLocated(HOW_MUCH_AND_PAY_INFORMATION));
+        boolean isDisplayedInformation = driver.findElement(HOW_MUCH_AND_PAY_INFORMATION).isDisplayed();
+        Assert.assertTrue(isDisplayedInformation);
```
Необходимо исправить: Не следует писать сложные методы и выполнять в них проверки. Такой код трудно читаем. Лучше писать маленькие методы, чтобы была возможность переиспользовать их в других методах. Тест должен состоять из шагов (методы для получения актуального результата) и проверки. Проверка должна быть в конце теста (метода тестового класса). После проверок и между проверками не должно быть дополнительных действий (шагов), иначе в случае фейла перовой проверки не будут выполнены остальные шаги, что может аффектить другие тесты.

nikolay-977 commented on src/main/java/ru/yandex/prakrikum/UserInformationForOrder.java:
```diff
@@ -0,0 +1,60 @@
+
+    //Метод заполнения формы Для кого самокат с первым набором данным
+    public void inputInformationAboutCustomerFirstDataSet(){
+        driver.findElement(NAME).clear();
```
Можно улучшить: В теме UI-тестирование > Page Object > Работа с методами > Объединение методов хорошо описан прием объединения методов. Применяй его для сокращения кода автотеста, где это необходимо. Например, при заполнении форм. Это позволит избежать повторяющегося кода.

nikolay-977 commented on src/main/java/ru/yandex/prakrikum/UserInformationForOrder.java:
```diff
@@ -0,0 +1,60 @@
+    //Метод заполнения формы Для кого самокат со вторым набором данным
+    public void inputInformationAboutCustomerSecondDataSet(){
+        driver.findElement(NAME).clear();
+        driver.findElement(NAME).sendKeys("Дмитрий");
```
Необходимо исправить: Перенеси тестовые данные в тесты, либо в отдельный класс для тестовых данных в src.tests.java.ru.yandex.praktikum.scooter.tests. Не следует хардкодить тестовые данные внутри методов. Тестовые данные следует передавать в качестве параметров при вызове методов. Это позволит избежать повторяющегося кода.

nikolay-977 commented on src/test/java/ru/yandex/praktikum/TestingImportantQuestion.java:
```diff
@@ -0,0 +1,41 @@
+
+public class TestingImportantQuestion {
+    private WebDriver driver;
+    public TestingImportantQuestion(){
```
Необходимо исправить: У каждого класса по умолчанию есть конструктор без аргументов, если нет других конструкторов. Конструктор без аргументов следует описывать, когда мы реализуем хотя бы один конструктор с аргументами.

firuzaminov committed:
Первый коммит клонированного репозитория

nikolay-977 commented on src/main/java/ru/yandex/prakrikum/UserInformationForOrder.java:
```diff
@@ -15,46 +17,44 @@ public class UserInformationForOrder {
-    public void inputInformationAboutCustomerSecondDataSet(){
+
+    //Метод заполнения формы Для кого самокат со вторым набором данным.
+    public void inputInformationAboutCustomerSecondDataSet(String nameFieldSecondDataSet, String surnameFieldSecondDataSet, String addressFieldSecondDataSet, String phoneNumberFieldSecondDataSet) {
```
Необходимо исправить: следует избегать повторяющегося кода. Вместо двух методов inputInformationAboutCustomerFirstDataSet() и inputInformationAboutCustomerSecondDataSet() следует реализовать один и назвать его inputInformationAboutCustomer(String name, String surname, String address, String phone). И вызывать его из тестов.

nikolay-977 commented on src/test/java/ru/yandex/praktikum/TestingScooterOrder.java:
```diff
@@ -1,49 +1,78 @@
+        rentInformation.clickOrderButton();
+        Assert.assertTrue(rentInformation.isDisplayedInformationSure());
+        rentInformation.clickApproveButtonYes();
+        Assert.assertTrue(rentInformation.isDisplayedSuccessWindow());
```
Необходимо исправить: Проверки в тестовых методах должны быть в одном стиле. Когда код написан в одном стиле, это улучшает читаемость кода, сокращает время локализации дефекта и упрощает его поддержку. Не следует в тестовых методах одновременно использовать проверки из разных фреймворков org.hamcrest.MatcherAssert(метод assertThat) и org.junit.Assert(методы assertEquals, assertTrue и т.д.). В этих фреймворках методы похожие, но имеют разные сигнатуры. Есть вероятность допустить ошибку.  В этих фреймворках отличается формат сообщений об ошибках в случае фейла, что не очень удобно при локализации дефектов.

firuzaminov committed:
Запараметризировал форму клиента (пока без кнопок)

firuzaminov committed:
Версия с кучей классов

firuzaminov committed:
Испраленная версия:
В классах Page Object теперь один метод для заполнения иннпутов
Заменени в тестовых классах Accert

firuzaminov committed:
Правки по комментариях ревьюера

firuzaminov committed:
Правки по комментариях ревьюера. Методы в RentInformation и в UserInformation были объденены в оди шаг. Исправлена проблема с Assert
